flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -335,14 +371,28 @@ class CodeGenerator(
         j"""
             |  public final void accumulate(
             |    org.apache.flink.types.Row accs,
-            |    org.apache.flink.types.Row input)""".stripMargin
+            |    org.apache.flink.types.Row input) throws Exception""".stripMargin
 
       val accumulate: String = {
         for (i <- aggs.indices) yield
-          j"""
+         if(distinctAggsFlags(i)){
+           j"""
+              |  Long distValCount$i = (Long) distStateList[$i].get(${parameters(i)});
+              |  if( distValCount$i == null){
```

Comment:
Ah, that makes sense. Scala `Long` is backed by Java `long`, i.e., it does not support `null`.